File: ios/ShortDrama/Sources/Domain/Entities/PlayerModels.swift
import Foundation

/// Series status metadata used by the player page.
enum DramaSeriesStatus: String, Codable, Equatable, Sendable {
    case completed
    case ongoing

    var displayText: String {
        switch self {
        case .completed:
            return "已完结"
        case .ongoing:
            return "更新中"
        }
    }
}

/// Episode list payload used by the player bootstrap flow.
struct EpisodeList: Equatable, Sendable {
    let dramaId: String
    let seriesStatus: DramaSeriesStatus
    let items: [Episode]
}

/// Resume progress metadata returned by the player progress API.
struct PlayerProgress: Equatable, Sendable {
    let dramaId: String
    let hasHistory: Bool
    let episodeId: String?
    let startTime: Double
    let updatedAt: String?
}

/// Request model for start playback.
struct StartPlaybackRequest: Equatable, Sendable {
    let dramaId: String
    let episodeId: String
    let progress: Double
}

/// Request model for stop playback.
struct StopPlaybackRequest: Equatable, Sendable {
    let dramaId: String
    let episodeId: String
    let progress: Double
    let duration: Double
}

/// Response model for a successful start playback call.
struct PlaybackStartReceipt: Equatable, Sendable {
    let dramaId: String
    let episodeId: String
    let acceptedProgress: Double
    let playbackSessionId: String
    let startedAt: String
}

/// Response model for a successful stop playback call.
struct PlaybackStopReceipt: Equatable, Sendable {
    let dramaId: String
    let episodeId: String
    let savedProgress: Double
    let duration: Double
    let updatedAt: String
}

File: android/app/src/main/java/com/djs66256/short_drama/domain/model/PlayerModels.kt
package com.djs66256.short_drama.domain.model

enum class SeriesStatus(
    val wireValue: String,
    val label: String,
) {
    COMPLETED(wireValue = "completed", label = "已完结"),
    ONGOING(wireValue = "ongoing", label = "更新中"),
    ;

    companion object {
        fun fromWireValue(value: String?): SeriesStatus {
            return entries.firstOrNull { it.wireValue == value } ?: COMPLETED
        }
    }
}

data class DramaEpisodeList(
    val dramaId: String,
    val seriesStatus: SeriesStatus,
    val items: List<Episode>,
)

data class PlaybackProgress(
    val dramaId: String,
    val hasHistory: Boolean,
    val episodeId: String?,
    val startTime: Double,
    val updatedAt: String?,
)

data class StartPlaybackParams(
    val dramaId: String,
    val episodeId: String,
    val progress: Double,
)

data class StartPlaybackResult(
    val dramaId: String,
    val episodeId: String,
    val acceptedProgress: Double,
    val playbackSessionId: String,
    val startedAt: String,
)

data class StopPlaybackParams(
    val dramaId: String,
    val episodeId: String,
    val progress: Double,
    val duration: Double,
)

data class StopPlaybackResult(
    val dramaId: String,
    val episodeId: String,
    val savedProgress: Double,
    val duration: Double,
    val updatedAt: String,
)
